src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/gencode/GencodeFuncotationFactory.java

```diff
@@ -2121,33 +2133,107 @@ private GencodeFuncotation createIgrFuncotation(final VariantContext variant,
+                .setEnd(variant.getEnd())
+                .setVariantType(getVariantType(variant.getReference(), altSymbolicAllele))
+                .setChromosome(variant.getContig())
+                .setOtherTranscripts(Collections.emptyList())
+                .setAnnotationTranscript(annotationTranscript.getTranscriptId());
+
+        funcotationBuilder.setHugoSymbol( annotationTranscript.getGeneName() );
+
+        // If we have a cached value for the ncbiBuildVersion, we should add it:
+        // NOTE: This will only be true if we have previously annotated a non-IGR variant.
+        // TODO: This is issue #4404
+        if ( ncbiBuildVersion != null ) {
+            funcotationBuilder.setNcbiBuild( ncbiBuildVersion );
+        }
+
+        final StrandCorrectedReferenceBases referenceBases = FuncotatorUtils.getBasesInWindowAroundReferenceAllele(variant.getReference(), reference, Strand.POSITIVE, referenceWindow);
+
+        // Set our reference context in the the FuncotatonBuilder:
+        funcotationBuilder.setReferenceContext( referenceBases.getBaseString() );
+
+        // Set our version:
+        funcotationBuilder.setVersion(version);
+
+        // Set our data source name:
+        funcotationBuilder.setDataSourceName(getName());
+
+        return funcotationBuilder.build();
+    }
+
+    /**
+     * Creates a {@link GencodeFuncotation} based on a given variant and alt allele.
+     * Either the reference or alternate allele contains at least one masked base.
      *
      * Reports reference bases as if they are on the {@link Strand#POSITIVE} strand.
      * @param variant The {@link VariantContext} associated with this annotation.
-     * @param annotationTranscript The transcript ID to use for populating this funcotation.
+     * @param altAllele The alternate {@link Allele} associated with this annotation.
+     * @param annotationTranscript The transcript in which this variant occurs.
      * @param reference The {@link ReferenceContext} in which the given {@link Allele}s appear.
      * @return An IGR funcotation for the given allele.
      */
-    private GencodeFuncotation createFuncotationForSpanningDeletion(final VariantContext variant,
-                                                                    final String annotationTranscript,
-                                                                    final ReferenceContext reference){
+    private GencodeFuncotation createFuncotationForMaskedBases(final VariantContext variant,
+                                                             final Allele altAllele,
+                                                             final GencodeGtfTranscriptFeature annotationTranscript,
+                                                             final ReferenceContext reference) {
+
+        final StringBuilder alleleLogStringBuilder = new StringBuilder();
+        if ( variant.getReference().getBaseString().contains("N") ) {
```

Comment:
Magic constant.  I bet there is already a constant defined for "N".  Even if not, it should not be inline here.

Reply:
There doesn't seem to be one.  It should really live in `Allele.java`.  I'll add an issue to HTSJDK I guess.

In the meantime, I've created my own that will need to be removed upon completion of that issue.

Reply:
Added in HTSJDK as issue 1226 (https://github.com/samtools/htsjdk/issues/1226)